audiostream/CMakeLists.txt

```diff
@@ -318,6 +318,11 @@ target_link_libraries       ( audiostream PUBLIC pcm
 
 if(MSVC)
   target_compile_options( audiostream PRIVATE /wd4351 )  # MSVC new behaviour warning
+  
+  # https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4103?view=vs-2019
+  # C4103 alignment changed after including header, may be due to missing #pragma pack(pop)
+  # C5038 data member 'SyncronousByteStream<Source>::ReadResult::m_bytesRead' will be initialized after data member 'SyncronousByteStream<Source>::ReadResult::m_references'
```

Comment:
Would you be up for fixing these warnings instead of asking MSVC to ignore them?  In the case of C5038, you just need to swap the member variable orders around.

Reply:
Yes the C5038 one is trivial to fix.

I've investigated C4103 a bit and did not find a proper solution.
Apparently it's related to boost using `#pragma` directives regarding alignment in separate headers. I'm not familiar enough with the ni-media sources to understand where that's coming from.


Can we split the PR in two, as fixing the compile errors is most important ?
Warnings one can easily suppress those in a parent `CMakeLists.txt` as a workaround.

Reply:
Some context regarding C4103:

https://stackoverflow.com/questions/403946/trying-to-get-rid-of-a-c-boost-warning
https://lists.boost.org/Archives/boost/2008/09/142674.php

To summarize, it's MSVC being over pedantic in this case, and it should be safe to disable the warning in ni_media.

Reply:
I'd be fine with splitting the PR into two in order to tackle the compile errors first.

Reply:
@monsieurnoir Yes, I've reverted the CMakeLists.txt changes pertaining to warnings suppression.